diff -Nur nautilus/src/Makefile.am nautilus_new/src/Makefile.am
--- nautilus/src/Makefile.am	2017-02-24 13:09:28.733348092 +0000
+++ nautilus_new/src/Makefile.am	2017-02-24 13:31:35.150034439 +0000
@@ -125,6 +125,10 @@
 	nautilus-batch-rename-dialog.h \
 	nautilus-batch-rename-utilities.c \
 	nautilus-batch-rename-utilities.h
+else
+nautilus_batch_renaming_tracker_sources = \
+	nautilus-batch-rename-utilities.c \
+	nautilus-batch-rename-utilities.h
 endif
 
 nautilus_built_sources = \
diff -Nur nautilus/src/nautilus-batch-rename-utilities.c nautilus_new/src/nautilus-batch-rename-utilities.c
--- nautilus/src/nautilus-batch-rename-utilities.c	2017-02-24 13:09:28.736681426 +0000
+++ nautilus_new/src/nautilus-batch-rename-utilities.c	2017-02-24 13:30:24.256700054 +0000
@@ -729,6 +729,7 @@
     return selection;
 }
 
+#ifdef ENABLE_TRACKER
 static void
 cursor_next (QueryData           *query_data,
              TrackerSparqlCursor *cursor)
@@ -738,6 +739,7 @@
                                       on_cursor_callback,
                                       query_data);
 }
+#endif
 
 static void
 remove_metadata (QueryData    *query_data,
@@ -778,6 +780,7 @@
     return formated_date;
 }
 
+#ifdef ENABLE_TRACKER
 static void
 on_cursor_callback (GObject      *object,
                     GAsyncResult *result,
@@ -1119,6 +1122,7 @@
     g_object_unref (connection);
     g_string_free (query, TRUE);
 }
+#endif
 
 GList *
 batch_rename_files_get_distinct_parents (GList *selection)
diff -Nur nautilus/src/nautilus-batch-rename-utilities.h nautilus_new/src/nautilus-batch-rename-utilities.h
--- nautilus/src/nautilus-batch-rename-utilities.h	2017-02-24 13:09:28.626681424 +0000
+++ nautilus_new/src/nautilus-batch-rename-utilities.h	2017-02-24 13:24:58.903361893 +0000
@@ -21,7 +21,11 @@
 
 #include <gio/gio.h>
 #include <gtk/gtk.h>
+
+#include <config.h>
+#ifdef ENABLE_TRACKER
 #include <tracker-sparql.h>
+#endif
 
 GList* batch_rename_dialog_get_new_names_list          (NautilusBatchRenameDialogMode  mode,
                                                         GList                         *selection,
