diff -Nur gnome-control-center-3.18.2/config.h.in gnome-control-center-3.18.2_new/config.h.in
--- gnome-control-center-3.18.2/config.h.in	2016-01-09 08:42:11.507082612 +0000
+++ gnome-control-center-3.18.2_new/config.h.in	2016-01-09 08:41:02.893748255 +0000
@@ -18,6 +18,9 @@
 /* Define if debugging is enabled */
 #undef GNOME_ENABLE_DEBUG
 
+/* Define to 1 to build the smbclient-printer panel */
+#undef HAVE_SMBCLIENT
+
 /* Define to 1 to build the Colord panel */
 #undef HAVE_COLORD
 
diff -Nur gnome-control-center-3.18.2/configure.ac gnome-control-center-3.18.2_new/configure.ac
--- gnome-control-center-3.18.2/configure.ac	2016-01-09 08:29:36.087071401 +0000
+++ gnome-control-center-3.18.2_new/configure.ac	2016-01-09 08:41:10.417081706 +0000
@@ -252,7 +252,12 @@
 
 if test x"$have_smbclient" = x"no"; then
     AC_MSG_WARN(*** Printer panel will not be built (smbclient.pc not found) ***)
-    enable_cups=no
+    # enable_cups=no
+    AC_DEFINE(BUILD_SMBCLIENT, 0, [Define to 1 to build the smbclient-printer panel])
+else
+    AC_DEFINE(BUILD_SMBCLIENT, 1, [Define to 1 to build the smbclient-printer panel])
+    AC_DEFINE(HAVE_SMBCLIENT, 1, [Define to 1 to build the smbclient-printer panel])
 fi
+AM_CONDITIONAL(BUILD_SMBCLIENT, [test x$have_smbclient = xyes])
 
 # Check for CUPS 1.4 or newer
diff -Nur gnome-control-center-3.18.2/panels/printers/Makefile.am gnome-control-center-3.18.2_new/panels/printers/Makefile.am
--- gnome-control-center-3.18.2/panels/printers/Makefile.am	2015-09-21 14:59:31.000000000 +0000
+++ gnome-control-center-3.18.2_new/panels/printers/Makefile.am	2016-01-09 08:38:32.863746033 +0000
@@ -41,13 +41,17 @@
 	pp-jobs-dialog.h		\
 	pp-authentication-dialog.c	\
 	pp-authentication-dialog.h	\
-	pp-samba.c			\
-	pp-samba.h			\
 	pp-print-device.c		\
 	pp-print-device.h		\
 	cc-printers-panel.c		\
 	cc-printers-panel.h
 
+if BUILD_SMBCLIENT
+libprinters_la_SOURCES +=		\
+	pp-samba.c			\
+	pp-samba.h
+endif
+
 libprinters_la_CPPFLAGS = $(AM_CPPFLAGS)
 libprinters_la_LIBADD = $(PRINTERS_PANEL_LIBS) $(PANEL_LIBS) $(CUPS_LIBS) $(SMBCLIENT_LIBS)
 
diff -Nur gnome-control-center-3.18.2/panels/printers/pp-new-printer-dialog.c gnome-control-center-3.18.2_new/panels/printers/pp-new-printer-dialog.c
--- gnome-control-center-3.18.2/panels/printers/pp-new-printer-dialog.c	2015-09-21 14:59:31.000000000 +0000
+++ gnome-control-center-3.18.2_new/panels/printers/pp-new-printer-dialog.c	2016-01-09 09:42:41.563803136 +0000
@@ -34,7 +34,9 @@
 #include "pp-utils.h"
 #include "pp-host.h"
 #include "pp-cups.h"
+#ifdef HAVE_SMBCLIENT
 #include "pp-samba.h"
+#endif
 #include "pp-new-printer.h"
 
 #include <gdk/gdkx.h>
@@ -127,7 +129,9 @@
   PpHost  *socket_host;
   PpHost  *lpd_host;
   PpHost  *remote_cups_host;
+#ifdef HAVE_SMBCLIENT
   PpSamba *samba_host;
+#endif
   guint    host_search_timeout_id;
 };
 
@@ -248,6 +252,7 @@
   gpointer  dialog;
 } AuthSMBData;
 
+#ifdef HAVE_SMBCLIENT
 static void
 get_authenticated_samba_devices_cb (GObject      *source_object,
                                     GAsyncResult *res,
@@ -367,6 +372,7 @@
         }
     }
 }
+#endif
 
 static void
 pp_new_printer_dialog_init (PpNewPrinterDialog *dialog)
@@ -416,7 +422,9 @@
   g_signal_connect (widget, "search-changed", G_CALLBACK (search_entry_changed_cb), dialog);
 
   widget = WID ("authenticate-button");
+#ifdef HAVE_SMBCLIENT
   g_signal_connect (widget, "clicked", G_CALLBACK (authenticate_samba_server), dialog);
+#endif
 
   /* Set junctions */
   widget = WID ("scrolledwindow1");
@@ -721,7 +729,9 @@
               priv->snmp_host != NULL ||
               priv->socket_host != NULL ||
               priv->lpd_host != NULL ||
+#ifdef HAVE_SMBCLIENT
               priv->samba_host != NULL ||
+#endif
               priv->samba_authenticated_searching ||
               priv->samba_searching;
 
@@ -1142,6 +1152,7 @@
     }
 }
 
+#ifdef HAVE_SMBCLIENT
 static void
 get_samba_host_devices_cb (GObject      *source_object,
                            GAsyncResult *res,
@@ -1235,6 +1246,7 @@
       g_error_free (error);
     }
 }
+#endif
 
 static void
 get_jetdirect_devices_cb (GObject      *source_object,
@@ -1446,8 +1458,10 @@
         g_object_set (priv->lpd_host, "port", data->host_port, NULL);
     }
 
+#ifdef HAVE_SMBCLIENT
   priv->samba_host = pp_samba_new (GTK_WINDOW (priv->dialog),
                                    data->host_name);
+#endif
 
   update_dialog_state (data->dialog);
 
@@ -1471,11 +1485,13 @@
                                  get_lpd_devices_cb,
                                  data->dialog);
 
+#ifdef HAVE_SMBCLIENT
   pp_samba_get_devices_async (priv->samba_host,
                               TRUE,
                               priv->remote_host_cancellable,
                               get_samba_host_devices_cb,
                               data->dialog);
+#endif
 
   priv->host_search_timeout_id = 0;
 
@@ -1894,7 +1910,9 @@
 {
   PpNewPrinterDialogPrivate *priv = dialog->priv;
   GtkTreeViewColumn         *column;
+#ifdef HAVE_SMBCLIENT
   PpSamba                   *samba;
+#endif
   GEmblem                   *emblem;
   PpCups                    *cups;
   GIcon                     *icon, *emblem_icon;
@@ -1944,8 +1962,10 @@
   priv->samba_searching = TRUE;
   update_dialog_state (dialog);
 
+#ifdef HAVE_SMBCLIENT
   samba = pp_samba_new (GTK_WINDOW (priv->dialog), NULL);
   pp_samba_get_devices_async (samba, FALSE, priv->cancellable, get_samba_devices_cb, dialog);
+#endif
 }
 
 static void
