diff -Nur gnome-control-center-3.18.2/config.h.in gnome-control-center-3.18.2_new/config.h.in
--- gnome-control-center-3.18.2/config.h.in	2016-01-09 08:03:15.000000000 +0000
+++ gnome-control-center-3.18.2_new/config.h.in	2016-01-09 08:11:33.190388668 +0000
@@ -18,6 +18,9 @@
 /* Define if debugging is enabled */
 #undef GNOME_ENABLE_DEBUG
 
+/* Define to 1 to build the Colord panel */
+#undef HAVE_COLORD
+
 /* Define to 1 if OnlineAccounts support is available */
 #undef HAVE_ONLINE_ACCOUNTS
 
diff -Nur gnome-control-center-3.18.2/configure.ac gnome-control-center-3.18.2_new/configure.ac
--- gnome-control-center-3.18.2/configure.ac	2016-01-09 08:03:03.983714447 +0000
+++ gnome-control-center-3.18.2_new/configure.ac	2016-01-09 08:10:38.270387848 +0000
@@ -154,7 +154,16 @@
                   colord >= $COLORD_REQUIRED_VERSION
                   colord-gtk >= $COLORD_GTK_REQUIRED_VERSION
                   libsoup-2.4
-                  gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION)
+                  gnome-desktop-3.0 >= $GNOME_DESKTOP_REQUIRED_VERSION,
+                  [have_colord=yes], have_colord=no)
+if test "x$have_colord" = xyes ; then
+	AC_DEFINE(BUILD_COLORD, 1, [Define to 1 to build the Colord panel])
+fi
+AM_CONDITIONAL(BUILD_COLORD, [test x$have_colord = xyes])
+if test x${have_colord} = xyes; then
+  AC_DEFINE(HAVE_COLORD, 1, [Define to 1 if Colord support is available])
+fi
+
 PKG_CHECK_MODULES(PRINTERS_PANEL, $COMMON_MODULES
                   polkit-gobject-1 >= $POLKIT_REQUIRED_VERSION)
 PKG_CHECK_MODULES(PRIVACY_PANEL, $COMMON_MODULES)
diff -Nur gnome-control-center-3.18.2/panels/Makefile.am gnome-control-center-3.18.2_new/panels/Makefile.am
--- gnome-control-center-3.18.2/panels/Makefile.am	2016-01-09 08:03:03.983714447 +0000
+++ gnome-control-center-3.18.2_new/panels/Makefile.am	2016-01-09 08:07:30.190385064 +0000
@@ -2,7 +2,6 @@
 	common \
 	background \
 	power \
-	color \
 	display \
 	mouse \
 	notifications \
@@ -21,6 +21,10 @@
 SUBDIRS += online-accounts
 endif
 
+if BUILD_COLORD
+SUBDIRS += color
+endif
+
 if BUILD_WACOM
 SUBDIRS += wacom
 endif
diff -Nur gnome-control-center-3.18.2/shell/cc-panel-loader.c gnome-control-center-3.18.2_new/shell/cc-panel-loader.c
--- gnome-control-center-3.18.2/shell/cc-panel-loader.c	2016-01-09 08:03:03.983714447 +0000
+++ gnome-control-center-3.18.2_new/shell/cc-panel-loader.c	2016-01-09 08:08:26.947052572 +0000
@@ -33,7 +33,9 @@
 #ifdef BUILD_BLUETOOTH
 extern GType cc_bluetooth_panel_get_type (void);
 #endif /* BUILD_BLUETOOTH */
+#ifdef BUILD_COLORD
 extern GType cc_color_panel_get_type (void);
+#endif /* BUILD_COLORD */
 extern GType cc_date_time_panel_get_type (void);
 extern GType cc_display_panel_get_type (void);
 extern GType cc_info_panel_get_type (void);
@@ -79,7 +81,9 @@
 #ifdef BUILD_BLUETOOTH
   PANEL_TYPE("bluetooth",        cc_bluetooth_panel_get_type    ),
 #endif
+#ifdef BUILD_COLORD
   PANEL_TYPE("color",            cc_color_panel_get_type        ),
+#endif
   PANEL_TYPE("datetime",         cc_date_time_panel_get_type    ),
   PANEL_TYPE("display",          cc_display_panel_get_type      ),
   PANEL_TYPE("info",             cc_info_panel_get_type         ),
diff -Nur gnome-control-center-3.18.2/shell/Makefile.am gnome-control-center-3.18.2_new/shell/Makefile.am
--- gnome-control-center-3.18.2/shell/Makefile.am	2016-01-09 09:05:22.393769913 +0000
+++ gnome-control-center-3.18.2_new/shell/Makefile.am	2016-01-09 09:10:14.583774246 +0000
@@ -58,7 +58,6 @@
 	$(top_builddir)/panels/common/liblanguage.la			\
 	$(top_builddir)/panels/common/libdevice.la			\
 	$(top_builddir)/panels/background/libbackground.la		\
-	$(top_builddir)/panels/color/libcolor.la			\
 	$(top_builddir)/panels/datetime/libdate_time.la			\
 	$(top_builddir)/panels/display/libdisplay.la			\
 	$(top_builddir)/panels/info/libinfo.la				\
@@ -74,6 +73,10 @@
 	$(top_builddir)/panels/universal-access/libuniversal-access.la	\
 	$(top_builddir)/panels/user-accounts/libuser-accounts.la
 
+if BUILD_COLORD
+gnome_control_center_LDADD += $(top_builddir)/panels/color/libcolor.la
+endif
+
 if BUILD_ONLINE_ACCOUNTS
 gnome_control_center_LDADD += $(top_builddir)/panels/online-accounts/libonline-accounts.la
 endif
