diff -Nur old/src/caja-python.c new/src/caja-python.c
--- old/src/caja-python.c	2015-05-08 06:33:48.000000000 +0000
+++ new/src/caja-python.c	2015-08-09 10:13:22.189518411 +0000
@@ -161,8 +161,8 @@
 	if (Py_IsInitialized())
 		return TRUE;
 
-  	debug("g_module_open " PY_LIB_LOC "/libpython" PYTHON_VERSION "." G_MODULE_SUFFIX ".1.0");
-	libpython = g_module_open(PY_LIB_LOC "/libpython" PYTHON_VERSION "." G_MODULE_SUFFIX ".1.0", 0);
+  	debug("g_module_open "  "libpython" PYTHON_VERSION "." G_MODULE_SUFFIX );
+	libpython = g_module_open( "libpython" PYTHON_VERSION "." G_MODULE_SUFFIX , 0);
 	if (!libpython)
 		g_warning("g_module_open libpython failed: %s", g_module_error());
 
