diff -Nur codelite-9.2_old/WordCompletion/WordTokenizer.cpp codelite-9.2/WordCompletion/WordTokenizer.cpp
--- codelite-9.2_old/WordCompletion/WordTokenizer.cpp	2016-07-07 05:54:58.000000000 +0000
+++ codelite-9.2/WordCompletion/WordTokenizer.cpp	2016-09-30 02:45:28.525280000 +0000
@@ -2178,7 +2178,7 @@
     wordlex_init(&scanner);
     struct yyguts_t * yyg = (struct yyguts_t*)scanner;
     //wxScopedCharBuffer cb = buffer.mb_str(wxConvUTF8);
-    word_switch_to_buffer(word_scan_string(buffer,scanner),scanner);
+    word_switch_to_buffer(word_scan_string(buffer.c_str(),scanner),scanner);
     yycolumn = 1;
     return scanner;
 }
