diff -Nur lximage-qt-1.3.0/src/application.cpp lximage-qt-1.2.0_new/src/application.cpp
--- lximage-qt-1.3.0/src/application.cpp	2022-11-05 16:10:14.000000000 +0800
+++ lximage-qt-1.3.0_new/src/application.cpp	2023-05-21 10:31:34.993306654 +0800
@@ -54,7 +54,18 @@
   installTranslator(libFm.translator());
 
   // install our own tranlations
-  translator.load(QStringLiteral("lximage-qt_") + QLocale::system().name(), QStringLiteral(LXIMAGE_DATA_DIR) + QStringLiteral("/translations"));
+  if (!translator.load(QStringLiteral("lximage-qt_") + QLocale::system().name(), QStringLiteral(LXIMAGE_DATA_DIR) + QStringLiteral("/translations")))
+  {
+    QStringList dirs;
+    dirs << QString::fromLocal8Bit(qgetenv("XDG_DATA_DIRS")).split(QLatin1Char(':'), Qt::SkipEmptyParts);
+    for(const QString &p : qAsConst(dirs))
+    {
+        if (translator.load(QLatin1String("lximage-qt_") + QLocale::system().name(), p + QLatin1String("/lximage-qt/translations")))
+        {
+            break;
+        }
+    }
+  }
   installTranslator(&translator);
 
   // initialize dbus
